### PR TITLE
Support primitive types array declaration

### PR DESCRIPTION
**Pull request by AhernEYeldarB**
48ec1f268ce9bc9fa89b653a603132a5f9e063a5 commit adds support for arrays as we had previously worked on them. When using assignment we need to check
* The expression is an array if an array is declared
* The expression array has types that are of the type declared

0e0790664b12346383a2375a7a5f53daf9ef2bb0 init constant file types for accessibility/readability purposes not necessary

Not sure exactly where we left off here. Suprised TypeScript still exists. **_Gonna commit most recent package files just incase dont let the +1,907 −1,529 fool you_**. God forbid anything before 5ea68120395f8c5bab44303fee66fc98df4ac5a9.
Looks like structure is dead/not implemented ATM 958c7630e0ce370a8e504f2d407f85c284ca1ab9 as sometimes accessed as undefined. 

_Outside the scope of this PR_ but;
* [This](https://github.com/AhernEYeldarB/vscode-hoshie/blob/121c31c3aed0e5c0a8e956d763f12ee763253561/src/grammar/syntaxVisitor.ts#L153) can return the first offending type in the array

Can add screenshots/gif if necessary, not sure of your expected PR flow here :))

### CODE REVIEW

File: src/constants/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  array: 'array',
+  string: 'string',
+  structure: 'structure',
+  number: 'number',
+}
+  
+
+export {
+  dataTypes,
+}
```
Comment (GordonSmith) [inline src/constants/types.ts:12]: Minor:  The following is more maintainable IMO
```
export const dataTypes = {
  boolean: 'boolean',
  array: 'array',
  string: 'string',
  structure: 'structure',
  number: 'number',
}
```

File: src/grammar/syntaxVisitor.ts [hunk 2]
```diff
@@ -139,18 +136,19 @@ export class SyntaxVisitor extends hoshieParser.getBaseCstVisitorConstructorWith
                 })
             }
         }
-        else if (assign && declaration && expression && (declaration.typeOf() !== expression.typeOf() || (declaration.isArray === true && expression.typeOf() !== "array"))) {
+        else if (assign && declaration && expression && ((!declaration.isArray && declaration.typeOf() !== expression.typeOf()) || (declaration.isArray === true && expression.typeOf() !== dataTypes.array))) {
```
Comment (GordonSmith) [inline src/grammar/syntaxVisitor.ts:139]: In general 
```
  Array.isArray(declaration)
```
is preferable to `declaration.isArray`

**Review by GordonSmith — CHANGES_REQUESTED** [review]
@AhernEYeldarB - I will ask Alex to make his final year project as public as the entire implementation is cleaner and more "correct" IMO - (primarily by using Antlr instead of chevrotain)

**Comment by AhernEYeldarB** [comment]
Coolio, @GordonSmith Will see what needs to be done on Antlr side if anything. I forgot things had changed for the fyp 🤦

**Comment by GordonSmith** [comment]
You can now fork it from here:  @AhernEYeldarB https://github.com/AlexSmith/hoshie-lang

**Pull request closed without merging**